frontend/src/app/features/team-planner/team-planner/planner/team-planner.component.ts

```diff
@@ -601,25 +601,30 @@ export class TeamPlannerComponent extends UntilDestroyedMixin implements OnInit,
     }
   }
 
-  isWpDateInCurrentView(workPackage:WorkPackageResource, date:'start'|'end'):boolean {
-    if (workPackage.startDate && workPackage.dueDate) {
-      let dateToCheck;
+  isWpEndDateInCurrentView(workPackage:WorkPackageResource):boolean {
+    const { dueDate } = workPackage;
 
-      const currentStartDate = this.ucCalendar.getApi().view.currentStart.setHours(0, 0, 0, 0);
-      const currentEndDate = this.ucCalendar.getApi().view.currentEnd.setHours(0, 0, 0, 0);
+    if (!dueDate) {
+      return !!workPackage.date;
+    }
+
+    const viewEndDate = this.ucCalendar.getApi().view.currentEnd.setHours(0, 0, 0, 0);
+    const dateToCheck = new Date(dueDate).setHours(0, 0, 0, 0);
 
-      if (date === 'start') {
-        dateToCheck = new Date(workPackage.startDate).setHours(0, 0, 0, 0);
-      } else {
-        dateToCheck = new Date(workPackage.dueDate).setHours(0, 0, 0, 0);
-      }
+    return dateToCheck < viewEndDate;
```

Comment:
This should be `=<`

Reply:
No, I'm not sure why exactly, but the `viewEndDate` is the date past the end of the calendar.